### PR TITLE
Encourage users to create accounts during workshop registration

### PR DESCRIPTION
**Pull request by jeremydstone**
This PR implements encouragement of Code.org account creation during workshop registration.

Part 1: on the workshop registration page itself, if the user is not signed in, the teal information box with ("already have a Code.org account?") is displayed. If the user is signed in, this information box is not shown. If the user clicks the "Sign In" link, she is sent to the sign in page with `return_to` param set so she will return to this registration page after signing in.
image

Part 2: on the thank you page, if the registered email does not exist as a Code.org account (regardless of whether the user is signed in or not), the "Get a Head Start" section below will be shown to encourage the user to create an account. We want them to create an account under the email they will be using to teach, which may not be the email they use during the summer. If the email they gave us doesn't exist as an account, then it is worth showing them this message.
image

### CODE REVIEW

**Comment by aoby** [comment]
LGTM, nice & simple. Please add unit tests to verify the expected parts appear or don't appear in each scenario.

File: dashboard/app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -2730,3 +2730,27 @@ code {
     }
   }
 }
+
+.info-box-teal {
+  border: 1px solid $teal;
+  height: 46px;
+  margin-bottom: 10px;
+  .left {
```
Comment (breville) [inline dashboard/app/assets/stylesheets/application.scss]: The scoping of this CSS to `.info-box-teal` is good.  The one thing to check is that we don't have any other more-globally declared `.left` that affects this component, though if it's looking okay then it's probably fine.

File: dashboard/app/views/pd/workshop_enrollment/new.html.haml [hunk 2]
```diff
@@ -16,6 +16,17 @@
         = render partial: 'workshop_details'
         %h2
           Your Information
+        - unless @signed_in
```
Comment (breville) [inline dashboard/app/views/pd/workshop_enrollment/new.html.haml]: A lot of our HAML just checks for `current_user`.  Does that work?

File: dashboard/app/views/pd/workshop_enrollment/new.html.haml [hunk 3]
```diff
@@ -16,6 +16,17 @@
...
+          %div.info-box-teal
+            %div.left
+              =image_tag image_url("info.png")
```
Comment (breville) [inline dashboard/app/views/pd/workshop_enrollment/new.html.haml]: I wonder if we should have a more specific filename than `info`.  Something like `info_icon` maybe?

**Comment by breville** [comment]
- Does the logo look okay on retina?  I wonder if it should be higher-resolution for such users?
- The screenshot shows a clickable URL, but the code seems to use the better "here" instead?
- Are there existing tests that can test this?  It seems worth testing.
- I'm slightly confused by this comment: "We want them to create an account under the email they will be using to teach, which may not be the email they use during the summer."  Because it sounds like the text we present is explicitly encouraging them to create an email that they will use to teach?

File: dashboard/app/assets/stylesheets/application.scss (hunk 1 above)
Comment (jeremydstone) [reply dashboard/app/assets/stylesheets/application.scss]: Thanks, have given more unique names to `.left` and `.right`

File: dashboard/app/views/pd/workshop_enrollment/new.html.haml (hunk 2 above)
Comment (jeremydstone) [reply dashboard/app/views/pd/workshop_enrollment/new.html.haml]: Yes, that's better, done, thx

File: dashboard/app/views/pd/workshop_enrollment/new.html.haml (hunk 3 above)
Comment (jeremydstone) [reply dashboard/app/views/pd/workshop_enrollment/new.html.haml]: Good idea, changed

**Comment by jeremydstone** [comment]
Responding to the above questions:

- Not sure how that looks on retina, that is the asset we have for the moment. We can update as necessary.
- Screenshot is from the code; "here" is clickable
- Tests coming
- Sorry, comment was unclear. We have an ongoing problem where teachers have a work email they use, and a separate personal email they use for the summer. We are trying our best to get them to use a Code.org account under their work email at the workshop, so that we can track their subsequent teaching activity.

**Pull request merged**